lib/posix-procfs/README

```diff
@@ -0,0 +1,36 @@
+posix-procfs: implementation of the proc fs
+
+Currently the library offers support for 5 files:
+/proc/meminfo
+/proc/version
+/prov/cmdline
+/proc/filesystems
+/proc/mounts
+
+Prerequisites:
+In order to use the library one must include vfscore and ukstore.
+Furthermore in order to mount proc at /proc one must mount a root
+file system.
+
+Of the implemented files the first two have no further prerequisites,
+while the latter three require libnewlib for text formatting.
+
+Implementation:
+The library creates the information on demand, when reading from the
+files. This is done with the help of static getters from the ukstore 
+library.
+The filesystem is mounted at /proc, while the containing files are 
+linked to the interior root (/).
+In order to know which files are read and return the proper
+information we parse the input in the read function.
+
+Contributing:
+In order to further expand the filesystem and add more entries there
+are a few steps necessary for the current format:
+-> create a new file after mounting the root file in procfs
+-> populate a buffer with the needed information (it would be nice if
+it is especially done with the help of ukstore)
+-> make sure to properly parse the input in the procfs_read function
+
+An example of the a test application can be found here:
+https://github.com/mandrei12/procfs-app
```

Comment:
I would directly include an example as a snippet rather than as a separate external link.

```diff
- An example of the a test application can be found here:
- https://github.com/mandrei12/procfs-app
+ For example, you can read the entries in procfs with:
+ ```
+ #include <stdio.h>
+ #include <fcntl.h>
+ #include <errno.h>
+ #include <stdlib.h>
+ #include <unistd.h>
+ 
+ int main()
+ {
+ 	char buf[200];
+ 	int fd;
+ 	fd = open("/proc/meminfo", O_RDONLY);
+	if (fd > 0) {
+		printf("/proc/meminfo:\n");
+		read(fd, buf, 200);
+		printf("%s", buf);
+	}
+ ```
```